### PR TITLE
support operators with type hints

### PR DESCRIPTION
**Pull request by wenjun93**
This fix pipeline component parser failure for component `__init__` with type hints, default parameters assignment with empty str alongside `=`  can be fixed as well.

### How was this pull request tested?

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by kiersten-stokes** [comment]
Hi @wenjun93! Thank you so much for your contribution - this is an area of the code that we have been wanting to clean up.

I see that this PR is still a WIP, but I have a few suggestions that may or may not help when it's ready for review. I think the area that you want to be looking in is the `_parse_properties()` function around line 109. Here we check for the presence of default value assignment (`=`), but we don't make any checks for `:` and type hints like we should.

I would also recommend adding a test for this particular case since parsing can result in so many edge cases (as the existence of this PR itself proves 😄 ). Our current tests for this functionality are located in `elyra/tests/pipeline/airflow/test_component_parser_airflow.py:172-290`. These 3 tests check against a dummy operator files located in `elyra/tests/pipeline/resource/components` (`airflow_test_operator.py` and `airflow_test_operator_no_inputs.py`). If you're familiar with pytest and feel comfortable doing so, I think it would be worthwhile to create a new resource file - maybe something like `airflow_test_operator_type_hints.py` and add a test to `test_component_parser_airflow.py` as well. 

Feel free to reach out with any questions!

**Comment by wenjun93** [comment]
@kiersten-stokes thanks for the detailed hints! updated.

File: etc/generic/requirements-elyra.txt [hunk 1]
```diff
@@ -15,6 +15,7 @@ requests==2.25.1
 tornado==6.0.4
 traitlets==4.3.3
 urllib3==1.26.5
+astunparse==1.6.3
```
Comment (akchinSTC) [inline etc/generic/requirements-elyra.txt:18]: Probably dont need to change this requirements file since this is strictly for generic operators. 
Remove this dependency and adding it to the root `setup.py` should resolve the CI issue

Comment (wenjun93) [reply etc/generic/requirements-elyra.txt]: @akchinSTC  oh, I did put the requirement mistakenly, moved now

**Review by kiersten-stokes — APPROVED** [review]
LGTM! Very nice test case
